README.md
#Tru Strap 

## handsome-vagrant-docker feature branch

![dockers](https://libcom.org/files/Dockers%20-%20North%20End%20King's%20Lynn.jpg)

## What is this?
This is the MSM/TSM public version of the 'tru-strap' script from https://github.com/jimfdavies/tru-strap. We have made this repo public so that we can download and start 'tru-strapping' without needing any credentials.  

## How do I use it?
We run this version of tru-strap via Vagrant using Docker as the VM provider on a centos 6 linux platform and space has been provided for both MSM and TSM company groups.

The base image used is **pauldavidgilligan/docker-centos6-base-image** this provides:

* centos 6
* supervisord
* supervisorctl
* default dev-opts user with sudo

### Vagrant Part

For example to start the agg (Aggregation Services):
```
export TRUSTRAP_ACCOUNT=msm
export TRUSTRAP_USERBASE=gb
export TRUSTRAP_ENV=hvd
export TRUSTRAP_SERVICE=agg

git@github.com:pauldavidgilligan-msm/tru-strap.git

git checkout handsome-vagrant-docker

cd tru-strap/docker/msm/centos6/agg

vagrant up --no-parallel
```

Destroy the services with:
```
vagrant destroy
```

Access an individual docker container node with:
```
vagrant ssh <node name>
```

### Docker Part

See what is running with:
```
docker ps -a
```

Any stranded containers should be removed with:
```
docker rm -f <container ID>
```

Inspect docker logs:
```
docker logs <container ID>
```

List docker images:
```
docker images
```

### Install
This will run both network and business services under a common docker host that is configured to create a VM under Oracle's Virtual Box. Although it is possible to run this vagrant project under boot2docker on Mac OS X and windows it is recommended that you simply install an Oracle Virtual Box VM with centos 6.6 64bit iso image and run the project there. Some issues with the docker autofs limits have been seen under MAC OS X.

* http://isoredirect.centos.org/centos/6/isos/x86_64/
* https://www.virtualbox.org/wiki/Downloads
* http://www.liquidweb.com/kb/how-to-install-docker-on-centos-6/

After ensuring the the VM's network is running and installing vagrant the general steps are:

```
su -

yum install epel-release -y

yum update

rpm -ivh https://dl.bintray.com/mitchellh/vagrant/vagrant_1.7.2_x86_64.rpm

yum erase docker
yum -y install docker-io
service docker start
chkconfig docker on
usermod -a -G docker dev-ops

docker search pauldavidgilligan

docker pull pauldavidgilligan/docker-centos6-base-image
```

### Docker Service

For private docker repos you will need to set the following:
```
vi /etc/sysconfig/docker
other_args=--insecure-registry=registry1-eu1.moneysupermarket.com:5000
```

### OS User ID

| Account       | ID            |
| ------------- |:-------------:|
| dev-ops       | 500 |
| jboss         | 600 |
| haproxy       | 610 |

### Issues
* https://github.com/docker/docker/issues/5135
* https://github.com/pauldavidgilligan-msm/go-skydns/issues/1
* https://github.com/pauldavidgilligan-msm/go-skydns/issues/2
* https://github.com/pauldavidgilligan-msm/go-skydns/issues/3

### Network Services

Network services such as skydns, loggly, haproxy and others are required to support the business services that we run under the docker host. Currently an image is available that combines both skydns and etcd on one docker image.

Etcd /skydns/config is used for the configuration for skydns rather than the environment, and during provisioning of the go-skydns node you will see for example:

```
==> go-skydns: Configuring skydns
==> go-skydns: {"action":"set","node":{"key":"/skydns/config","value":"{\"dns_addr\":\"0.0.0.0:53\",\"ttl\":3600, \"domain\":\"msm.internal\", \"nameservers\": [\"8.8.8.8:53\",\"8.8.4.4:53\"]}","modifiedIndex":3,"createdIndex":3}}
```
And then from a docker container as a business node you can test the dns with:

```
dig SRV agg-redis1.agg.dev.gb.msm.internal

[dev-opts@redis ~]$ dig SRV agg-redis1.agg.dev.gb.msm.internal

; <<>> DiG 9.8.2rc1-RedHat-9.8.2-0.30.rc1.el6 <<>> SRV agg-redis1.agg.dev.gb.msm.internal
;; global options: +cmd
;; Got answer:
;; ->>HEADER<<- opcode: QUERY, status: NOERROR, id: 51486
;; flags: qr aa rd ra; QUERY: 1, ANSWER: 1, AUTHORITY: 0, ADDITIONAL: 1

;; QUESTION SECTION:
;agg-redis1.agg.dev.gb.msm.internal. IN	SRV

;; ANSWER SECTION:
agg-redis1.agg.dev.gb.msm.internal. 17 IN SRV	10 100 80 n1.agg-redis1.agg.dev.gb.msm.internal.

;; ADDITIONAL SECTION:
n1.agg-redis1.agg.dev.gb.msm.internal. 17 IN A	172.17.0.8

```

### Business Services
These are defined in the services.yaml and the name represent not only the hostnames of the final docker containers but also the provisioning roles for puppet if required.

```
services:
  dev-agg:
    - domain: msm.internal
    - skydns: true
    - git:
      - reponame   : msm-provisioning
      - repouser   : pauldavidgilligan-msm
      - repobranch : handsome-vagrant-docker
    - roles:
      - data                        : agg-mongodb
        enabled                     : false
      - data                        : agg-lts-mongodb
        enabled                     : false
      - data                        : agg-redis
        enabled                     : false
      - app                         : agg-management-agent
        enabled                     : false
      - app                         : agg-private-api
        enabled                     : false
      - app                         : agg-public-api
        enabled                     : false
      - app                         : agg-performance-test-proxy
        enabled                     : false
      - app                         : agg_kafka
        enabled                     : false
```

* setting skydns to false allows the business containers to run alone for test purposes, normally due to the puppet provisioning both access to skydns and etcd are required.
* each role is then listened and the containers provisioned if enabled.

## Design Documentation
Main design ocumentation is maintained internally on confluence: 

https://moneysupermarket.atlassian.net/wiki/pages/viewpage.action?pageId=23921911

## Docker pull(s)

```
docker pull pauldavidgilligan/go-skydns:latest
docker pull registry1-eu1.moneysupermarket.com:5000/docker-centos6-base-image:latest
docker pull registry1-eu1.moneysupermarket.com:5000/docker-centos6-puppet-ruby215:latest
docker pull registry1-eu1.moneysupermarket.com:5000/docker-centos6-ejbca-mysql:latest
docker pull registry1-eu1.moneysupermarket.com:5000/docker-centos6-haproxy:latest
```
> Maintained in bin/images.sh

## Git Authentication
No provide keys are stored in the VM and ssh key forwarding is being used. You will need to fork
msm-provisioning, in this example, and setup you own keys on your local machine.

https://developer.github.com/guides/using-ssh-agent-forwarding/

## Environment Variables
This Vagrantfile requires a few environment variables to be set.

- ```export TRUSTRAP_ACCOUNT=msm``` The Account name.
- ```export TRUSTRAP_USERBASE=gb``` The User base.
- ```export TRUSTRAP_ENV=hvd```     The Environment(hvd for this project needed in msm-provisioning)
- ```export TRUSTRAP_SERVICE=agg``` The Business Service name.

## Testing

You can monitor progress from a business node for example:

```
[dev-opts@redis ~]$ tail -f /tmp/progress_2014_11_17_15.log  
2014-11-17 15:36:20 Docker Provision Start
2014-11-17 15:36:20 Verifying vagrant-shell
2014-11-17 15:36:20 Running vagrant-shell
2014-11-17 15:36:20 Installing RVM
2014-11-17 15:36:29 Installing RVM Ruby ruby-2.1.4
2014-11-17 15:40:16 Installing puppet tools
2014-11-17 15:40:41 Git: git clone --progress -b handsome-vagrant-docker git@github.com:pauldavidgilligan-msm/msm-provisioning.git /opt/msm-provisioning
2014-11-17 15:40:50 Installing puppet gems
2014-11-17 15:41:59 Installing trustrap puppet from /opt/msm-provisioning/puppet
2014-11-17 15:41:59 Installing trustrap puppet security from /opt/msm-provisioning/puppet
2014-11-17 15:42:00 Installing puppet role agg-redis
2014-11-17 15:42:56 Set Facter values
2014-11-17 15:42:56 Provison with puppet apply
2014-11-17 15:45:35 vagrant-shell Complete
```

### Test Run - go test -v ./... -race

From the go-skydns host you can run a self test:

```
export GOPATH=/go
go test -v ./... -race
dig @127.0.0.1  www.miek.nl
```
Then from one business node you can test DNS:
```
[dev-opts@redis ~]$  dig  www.miek.nl

; <<>> DiG 9.8.2rc1-RedHat-9.8.2-0.30.rc1.el6 <<>> www.miek.nl
;; global options: +cmd
;; Got answer:
;; ->>HEADER<<- opcode: QUERY, status: NOERROR, id: 21872
;; flags: qr rd ra; QUERY: 1, ANSWER: 2, AUTHORITY: 0, ADDITIONAL: 0

;; QUESTION SECTION:
;www.miek.nl.			IN	A

;; ANSWER SECTION:
www.miek.nl.		20893	IN	CNAME	a.miek.nl.
a.miek.nl.		21434	IN	A	176.58.119.54

;; Query time: 22 msec
;; SERVER: 8.8.8.8#53(8.8.8.8)
;; WHEN: Mon Nov 17 15:37:28 2014
;; MSG SIZE  rcvd: 61

```

## Trustrap Original

The original init.sh trustrap init.sh has been extended slightly but with the aim of maintaining compatability with the original and RightScale deployments.

## License

go-skydns is under the Apache 2.0 license. See the [LICENSE](LICENSE) file for details.



CHANGELOG.md
commit 812b8e19e7d8b457aca31d0f245c6a160cc0ade2
Author: Paul.Gilligan <Paul.Gilligan@moneysupermarket.com>
Date:   Tue Jan 27 16:23:39 2015 +0000

    dropwizard WIP + dns fixes

commit fc9d051bb5909238961bf53c644cf30ed5bb3a06
Author: Paul.Gilligan <Paul.Gilligan@moneysupermarket.com>
Date:   Tue Jan 27 13:09:25 2015 +0000

    add dns option + dropwizard WIP

commit c01fd2be180dd37021aa690bd27aded282096310
Author: Paul.Gilligan <Paul.Gilligan@moneysupermarket.com>
Date:   Mon Jan 26 16:41:52 2015 +0000

    dropwizard WIP

commit 3f70bd956e109a428a98a2aa09affc95bb51d2bf
Author: Paul.Gilligan <Paul.Gilligan@moneysupermarket.com>
Date:   Fri Jan 23 13:29:49 2015 +0000

    puppet WIP

commit 56c2758f64e4e3015eb7d3e3d4de867a87c684a4
Author: Paul.Gilligan <Paul.Gilligan@moneysupermarket.com>
Date:   Thu Jan 22 18:29:36 2015 +0000

    typo

commit b93f5c5cf9e5d12312d0811dea4a716942141028
Author: Paul.Gilligan <Paul.Gilligan@moneysupermarket.com>
Date:   Thu Jan 22 16:42:25 2015 +0000

    puppet fixes

commit a76869b7adf3162728aea2a2ccc8eaff79f8efa2
Author: Paul.Gilligan <Paul.Gilligan@moneysupermarket.com>
Date:   Thu Jan 22 15:34:35 2015 +0000

    add github key push

commit 1b055dd22f78b68be25c91e53c2eb96b5f00474d
Author: Paul Gilligan <Paul.Gillian@moneysupermarket.com>
Date:   Thu Jan 22 08:21:08 2015 +0000

    Update README.md

commit ad46008d0d65c9dc904f86239dda6fb3b5a18f9d
Author: Paul Gilligan <Paul.Gillian@moneysupermarket.com>
Date:   Thu Jan 22 08:09:44 2015 +0000

    Update README.md

commit dbab2bb2833a83b52435903343b21cba4a960e0f
Author: Paul.Gilligan <Paul.Gilligan@moneysupermarket.com>
Date:   Thu Jan 22 07:51:01 2015 +0000

    drop wizard WIP

commit 2709288c7382d41b52fb358f99a8643c73ae3d83
Author: Paul.Gilligan <Paul.Gilligan@moneysupermarket.com>
Date:   Wed Jan 21 17:24:56 2015 +0000

    drop wizard WIP

commit de384180b98c4174a93c62939f74761224fa088e
Author: Paul.Gilligan <Paul.Gilligan@moneysupermarket.com>
Date:   Wed Jan 21 15:28:25 2015 +0000

    update puppet

commit 9dd3360a6bc4b974d4e33ce0d6836fbe3805fb48
Author: Paul.Gilligan <Paul.Gilligan@moneysupermarket.com>
Date:   Tue Jan 20 17:07:31 2015 +0000

    toolbelt puppet WIP

commit 6b74b3f06b5411ccf59920a4ce41830c33d81e0b
Merge: 79f5729 b11775b
Author: Paul Gilligan <Paul.Gillian@moneysupermarket.com>
Date:   Tue Jan 20 13:54:37 2015 +0000

    Merge pull request #1 from chrisjomaron/handsome-vagrant-docker
    
    proxy configuration for when behind MSM firewall

commit 79f5729eb68207cd901e8c5ac517fde1b42dd9af
Author: Paul.Gilligan <Paul.Gilligan@moneysupermarket.com>
Date:   Tue Jan 20 13:49:37 2015 +0000

    docker pull

commit 17abc587ecd67c7c9e11dc659bbadb6705e994ba
Author: Paul.Gilligan <Paul.Gilligan@moneysupermarket.com>
Date:   Tue Jan 20 12:22:27 2015 +0000

    move images to internal repo

commit b11775bd23f7086e74afac97f8e9a24d254ed813
Author: dev-ops <chris.jomaron@moneysupermarket.com>
Date:   Mon Jan 19 22:41:48 2015 +0000

    proxy configuration for when behind MSM firewall

commit 76290df5ad2ed0a24b767b4170cd48782f177ba2
Author: Paul Gilligan <Paul.Gillian@moneysupermarket.com>
Date:   Mon Jan 19 17:05:25 2015 +0000

    Update /etc/sysconfig/docker

commit ae11b565ef46424e4b9ee5cf58230d25e0782d55
Author: Paul.Gilligan <Paul.Gilligan@moneysupermarket.com>
Date:   Mon Jan 19 15:12:45 2015 +0000

    adjust toolbelt::client::compiled

commit dfa2766129f573ef335a0c9fc1708e346f3500e0
Author: Paul.Gilligan <Paul.Gilligan@moneysupermarket.com>
Date:   Mon Jan 19 14:42:06 2015 +0000

    adjust toolbelt::client::compiled

commit c0d8aae75b0f838f92b37790b8ff5ebe040b526e
Author: Paul.Gilligan <Paul.Gilligan@moneysupermarket.com>
Date:   Thu Jan 15 09:28:58 2015 +0000

    puppet WIP

commit 3eae2cea8395fa81d7383d020807efe5c72c29df
Author: Paul.Gilligan <Paul.Gilligan@moneysupermarket.com>
Date:   Wed Jan 14 17:23:22 2015 +0000

    puppet WIP

commit 44029310cd107aa59730f919cda2a86323451271
Author: Paul.Gilligan <Paul.Gilligan@moneysupermarket.com>
Date:   Wed Jan 14 17:14:32 2015 +0000

    puppet WIP

commit 8b54a73f3c5cfcb4f9b5f4d2c359cd3ae3ccee8f
Author: Paul.Gilligan <Paul.Gilligan@moneysupermarket.com>
Date:   Wed Jan 14 17:03:45 2015 +0000

    puppet WIP

commit d9bfe63bbb6241456d458ee3ffa062f3d0ffb13e
Author: Paul.Gilligan <Paul.Gilligan@moneysupermarket.com>
Date:   Wed Jan 14 14:49:00 2015 +0000

    puppet WIP

commit 604df80187c710635db7b953c729827cdfee13a3
Author: Paul.Gilligan <Paul.Gilligan@moneysupermarket.com>
Date:   Wed Jan 14 14:18:33 2015 +0000

    puppet WIP

commit 915a6620916c19c84c4544f0e498d67477d0a6b8
Author: Paul.Gilligan <Paul.Gilligan@moneysupermarket.com>
Date:   Tue Jan 13 16:39:42 2015 +0000

    puppet WIP

commit a269b6f6cc000a53f195b541249cb360e0cd4b34
Author: Paul.Gilligan <Paul.Gilligan@moneysupermarket.com>
Date:   Mon Jan 12 16:21:42 2015 +0000

    puppet WIP

commit f3057381454143a7c3ed88a4a3767afc2de6c04f
Author: Paul.Gilligan <Paul.Gilligan@moneysupermarket.com>
Date:   Mon Jan 12 14:10:51 2015 +0000

    updated skydns_client

commit b404d6571034839369adabc0b539e9ebfc397f6d
Author: Paul.Gilligan <Paul.Gilligan@moneysupermarket.com>
Date:   Fri Jan 9 16:53:15 2015 +0000

    puppet WIP

commit 1deadcf100e4db8f005fa0ed39da728bf5623986
Author: Paul.Gilligan <Paul.Gilligan@moneysupermarket.com>
Date:   Fri Jan 9 16:09:19 2015 +0000

    puppet WIP

commit 19833608f65b1256839448490f71acd06b2f5066
Author: Paul.Gilligan <Paul.Gilligan@moneysupermarket.com>
Date:   Fri Jan 9 15:41:52 2015 +0000

    puppet WIP

commit 7ad0b8a056719c0c0362e15bf5361ed2893983c7
Author: Paul.Gilligan <Paul.Gilligan@moneysupermarket.com>
Date:   Thu Jan 8 16:42:22 2015 +0000

    add puppet maven/java

commit 5c5de4a3dc69ceca0a322222a3a509194b0bd422
Author: dev-ops <dev-ops@localhost.localdomain>
Date:   Mon Jan 5 16:42:29 2015 +0000

    add mysql user checkpoint

commit 3c47612866040a5e49c13e934206e00d46b84543
Author: Paul Gilligan <Paul.Gillian@moneysupermarket.com>

docker/centos6/agg/init.sh
#!/bin/bash
#
# init.sh       Provision MSM/TSM trustrap services.
#
# Authors:      Jim Davies, <Jim.Davies@moneysupermarket.com>
#               Paul Gilligan, <Paul.Gilligan@moneysupermarket.com>
#
# Description:  Install MSM/TSM based puppet provisioning software and 
#               provision the required puppet role. This script has been modified from the original concept to allow many applications
#               to be provisioned under a vagrant/docker hosted solution.
#
#               This has beed designed to be re-entrant, i.e. it can be re-run.    
# 
# Original:     https://github.com/MSMFG/tru-strap
# Git version:  https://github.com/pauldavidgilligan-msm/tru-strap
# Git branch:   handsome-vagrant-docker

# Usage:        ./init.sh -r agg-redis -e dev -u pauldavidgilligan-msm -n msm-provisioning -b handsome-vagrant-docker

VERSION=0.0.2
SCRIPTNAME=`basename $0`
RUBYVERSION="ruby-2.1.4"
NOW=$(date "+%Y_%m_%d_%H")
PROGRESS_LOG=/tmp/progress_${NOW}.log

echo $(date "+%Y-%m-%d %H:%M:%S") "Docker Provision Start" > ${PROGRESS_LOG}

# -----------------------------------------------------------------------------
# Functions
# -----------------------------------------------------------------------------
function _line {
  length=40
  printf -v line '%*s' "$length"
  echo ${line// /=}
}

function _bold {
  echo -e "\e[30;1m$1 \e[21m"
  echo $(date "+%Y-%m-%d %H:%M:%S") $1 >> ${PROGRESS_LOG}
}

function _err {
  length=40
  printf -v line '%*s' "$length"
  echo ${line// /-}
  echo -e "\e[31m$1 \e[39m"
  echo $(date "+%Y-%m-%d %H:%M:%S") $1 >> ${PROGRESS_LOG}
}

# functions
function usage {
cat <<EOF

    Usage: $0 [options]
    -h| --help             this usage text.
    -v| --version          the version.
    -r| --role             the trustrap role
    -e| --environment      the environment name.
    -u| --repouser         the git repository user name.
    -n| --reponame         the git repository name.
    -b| --repobranch       the git repository branch name.
    
EOF
}

function print_version {
  echo $1 $2
}

# -----------------------------------------------------------------------------
# Reset console
# -----------------------------------------------------------------------------
echo -e "\e[0m"
_line

# -----------------------------------------------------------------------------
# Process Command Line Params
# -----------------------------------------------------------------------------
while test -n "$1"; do
  case "$1" in
  --help|-h)
    usage
    exit
    ;;
  --version|-v)
    print_version $SCRIPTNAME $VERSION
    exit
    ;;
  --role|-r)
    TRUSTRAP_ROLE=$2
    shift
    ;;
  --environment|-e)
    TRUSTRAP_ENV=$2
    shift
    ;;
  --repouser|-u)
    TRUSTRAP_REPOUSER=$2
    shift
    ;;
  --reponame|-n)
    TRUSTRAP_REPONAME=$2
    shift
    ;;
  --repobranch|-b)
    TRUSTRAP_REPOBRANCH=$2
    shift
    ;;

  *)
    _err ", unknown argument: $1"
    usage
    exit 1
    ;;
  esac
  shift
done

# -----------------------------------------------------------------------------
# Check
# -----------------------------------------------------------------------------
_bold "Verifying ${SCRIPTNAME}"
if [[ ${TRUSTRAP_ROLE} == "" || ${TRUSTRAP_ENV} == "" || ${TRUSTRAP_REPOUSER} == "" || ${TRUSTRAP_REPONAME} == "" || ${TRUSTRAP_REPOBRANCH} == "" ]]; then
  _err ", missing argument(s)."
  usage
  exit 1
fi

# -----------------------------------------------------------------------------
# Run 
# -----------------------------------------------------------------------------
_bold "Running ${SCRIPTNAME}"
TRUSTRAP_REPODIR="/opt/${TRUSTRAP_REPONAME}"
if [[ -d "${TRUSTRAP_REPODIR}" ]]; then
  rm -rf "${TRUSTRAP_REPODIR}"
  echo "Removed previous git repostory ${TRUSTRAP_REPODIR}"
fi

_bold "Installing RVM"
curl -sSL https://get.rvm.io | bash -s stable 
source /etc/profile.d/rvm.sh
rvm reload

_bold "Installing Ruby ${RUBYVERSION}"
rvm install ${RUBYVERSION}
rvm use ${RUBYVERSION}

_bold "Installing puppet tools"
yum install -y http://yum.puppetlabs.com/puppetlabs-release-el-6.noarch.rpm 
yum install -y puppet 
yum install -y facter

GITCMD="git clone --progress -b ${TRUSTRAP_REPOBRANCH} git@github.com:${TRUSTRAP_REPOUSER}/${TRUSTRAP_REPONAME}.git ${TRUSTRAP_REPODIR}"
PUPPET_DIR="${TRUSTRAP_REPODIR}/puppet"
PUPPET_BASE_FILE="Puppetfile.base"
PUPPET_BASE_ROLE_FILE="Puppetfile.${TRUSTRAP_ROLE}"

if [[ ! -d /root/.ssh ]]; then
  echo "Add github.com to known_hosts"
  mkdir /root/.ssh && touch /root/.ssh/known_hosts && ssh-keyscan -H github.com >> /root/.ssh/known_hosts && chmod 600 /root/.ssh/known_hosts
fi
_bold "Git: ${GITCMD}"
`${GITCMD}`

_bold "Installing puppet gems"
gem install librarian-puppet --no-rdoc --no-ri 
gem install hiera-eyaml --no-ri --no-rdoc
gem install puppet --no-rdoc --no-ri

_bold "Installing trustrap puppet from ${PUPPET_DIR}"
rm -rf /etc/puppet ; ln -s ${PUPPET_DIR} /etc/puppet 
rm /etc/hiera.yaml ; ln -s /etc/puppet/hiera.yaml /etc/hiera.yaml

_bold "Installing trustrap puppet security from ${PUPPET_DIR}"
mkdir -p /etc/puppet/secure/keys
chmod 0500 /etc/puppet/secure/
chmod 0500 /etc/puppet/secure/keys
cd /etc/puppet/secure/
eyaml createkeys
# TODO: @Paul dev keys ?

_bold "Installing puppet role ${TRUSTRAP_ROLE}"
if [[ ! -f /etc/puppet/Puppetfiles/${PUPPET_BASE_FILE} ]] || [[ ! -f /etc/puppet/Puppetfiles/${PUPPET_BASE_ROLE_FILE} ]]; then
  _err "Error locating puppet role file /etc/puppet/Puppetfiles/${PUPPET_BASE_FILE}"
  _err "Error locating puppet role file /etc/puppet/Puppetfiles/${PUPPET_BASE_ROLE_FILE}"
  _err "Check files and role name, ${TRUSTRAP_ROLE}"
  exit 1
fi
cat /etc/puppet/Puppetfiles/${PUPPET_BASE_FILE} /etc/puppet/Puppetfiles/${PUPPET_BASE_ROLE_FILE} > /etc/puppet/Puppetfile
cd /etc/puppet
librarian-puppet install --verbose
librarian-puppet show

# -----------------------------------------------------------------------------
# Use RVM to revert Ruby version to back to system default (1.8.7)
# this is required due to changes in the yaml parser in ruby 2+
# -----------------------------------------------------------------------------
rvm --default use system
  
# -----------------------------------------------------------------------------
# Set factor values 
# TODO: @Jim some duplication!
# -----------------------------------------------------------------------------
_bold "Set Facter values"
mkdir -m 0600 -p /etc/facter/facts.d
echo "# trustrap generated custom values"       > /etc/facter/facts.d/init_custom_values.txt
echo "init_env=${TRUSTRAP_ENV}"                >> /etc/facter/facts.d/init_custom_values.txt
echo "init_role=${TRUSTRAP_ROLE}"              >> /etc/facter/facts.d/init_custom_values.txt
echo "init_repouser=${TRUSTRAP_REPOUSER}"      >> /etc/facter/facts.d/init_custom_values.txt
echo "init_reponame=${TRUSTRAP_REPONAME}"      >> /etc/facter/facts.d/init_custom_values.txt
echo "init_repobranch=${TRUSTRAP_REPOBRANCH}"  >> /etc/facter/facts.d/init_custom_values.txt

echo "msmid_env=${TRUSTRAP_ENV}"                >> /etc/facter/facts.d/init_custom_values.txt
echo "msmid_role=${TRUSTRAP_ROLE}"              >> /etc/facter/facts.d/init_custom_values.txt
echo "msmid_repouser=${TRUSTRAP_REPOUSER}"      >> /etc/facter/facts.d/init_custom_values.txt
echo "msmid_reponame=${TRUSTRAP_REPONAME}"      >> /etc/facter/facts.d/init_custom_values.txt
echo "msmid_repobranch=${TRUSTRAP_REPOBRANCH}"  >> /etc/facter/facts.d/init_custom_values.txt

# -----------------------------------------------------------------------------
# Pull the puppet string
# -----------------------------------------------------------------------------
_bold "Provison with puppet apply"
puppet apply --debug /etc/puppet/manifests/site.pp

_line
_bold "${SCRIPTNAME} Complete"
_line

